go/yaml_marshalling.md

```diff
@@ -0,0 +1,32 @@
+# YAML Marshalling
+
+There are multiple options to marshal yaml in Go but there is no built-in one.
+We decided to use https://github.com/ghodss/yaml library for that purpose.
+
+**NOTE:** The `!!binary` YAML tag does not work with this library. See
+https://github.com/ghodss/yaml#caveats.
+
+From https://godoc.org/github.com/ghodss/yaml:
+
+> In short, this package first converts YAML to JSON using go-yaml and then
+> uses json.Marshal and json.Unmarshal to convert to or from the struct. This
+> means that it effectively reuses the JSON struct tags as well as the custom
+> JSON methods MarshalJSON and UnmarshalJSON unlike go-yaml.
+
+There is a good explanation for doing so written down here:
+http://ghodss.com/2014/the-right-way-to-handle-yaml-in-golang.
```

Comment:
This URL doesn't work ATM. I hope this is temporary.

I created issue for that https://github.com/ghodss/yaml/issues/57.

Reply:
This works http://web.archive.org/web/20190603050330/http://ghodss.com/2014/the-right-way-to-handle-yaml-in-golang/ :)

Reply:
Thanks. Updated.